src/migrations/add-harvard-data/1682097649111-AddHarvardData.ts

```diff
@@ -0,0 +1,106 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import * as csvParser from 'csv-parser';
+import * as fs from 'fs';
+
+export class AddHarvardData1682097649111 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const stream = fs.createReadStream(__dirname + '/harvard.csv').pipe(
+      csvParser({
+        mapHeaders: ({ header }) => header.toLowerCase(),
+      }),
+    );
+
+    // datasets with unique values
+    const states = new Set();
+    const parties = new Set();
+    const candidates = new Set();
+    const elections = new Set();
+
+    // temporary variable, initial value
+    const stateElectionResults = {
+      year: 1976,
+      state: 'ALABAMA',
+      data: [],
+    };
+
+    for await (const row of stream) {
+      const party = row.party_simplified.replaceAll("'", "''");
+      const candidate = row.candidate.replaceAll("'", "''");
+
+      // Add unique state to db
+      if (!states.has(row.state)) {
```

Comment:
Let's use this approach instead of saving unique values in `Set`
```
// Add unique state to db
      await queryRunner.query(
        `INSERT INTO state (name)
        SELECT '${row.state}'
        WHERE NOT EXISTS (SELECT id FROM state WHERE name = '${row.state}')`
      );
```

Reply:
Done, but I left the EXCEPT operator. In the case of comparison by 2 or more fields, much less code is obtained.